net/utils.cpp

```diff
@@ -255,43 +257,46 @@ class DefaultResolver : public Resolver {
     ~DefaultResolver() { dnscache_.clear(); }
 
     IPAddr resolve(const char *host) override {
-        auto ctr = [&]() -> IPAddr * {
-            std::vector<IPAddr> addrs;
+        auto ctr = [&]() -> std::list<IPAddr>* {
+            auto addrs = new std::list<IPAddr>();
             photon::semaphore sem;
             std::thread([&]() {
-                int ret = gethostbyname(host, addrs);
-                if (ret < 0) {
-                    addrs.clear();
-                }
-                sem.signal(1);
+                auto now = std::chrono::steady_clock::now();
+                gethostbyname(host, *addrs);
+                auto time_elapsed = std::chrono::duration_cast<std::chrono::microseconds>(
+                                        std::chrono::steady_clock::now() - now).count();
+                if ((uint64_t)time_elapsed <= resolve_timeout_) sem.signal(1);
```

Comment:
timing control may not be precise enough?

Reply:
Cause the photon thread should sleep for more than resolve_timeout_, this is enought to run correctly, right?